api_tests/lib/omnilayer.js

```diff
@@ -0,0 +1,232 @@
+    if (!config) {
+        throw new Error("ledger.omnilayer configuration is needed");
+    }
+    return (_rpc_client =
+        _rpc_client ||
+        new BitcoinRpcClient({
+            network: "regtest",
+            port: config.rpc_port,
+            host: config.rpc_host,
+            username: config.rpc_username,
+            password: config.rpc_password,
+        }));
+}
+
+module.exports.getBalance = async function(tokenId, address) {
+    const res = await _rpc_client.command([
+        {
+            method: "omni_getbalance",
+            parameters: [address, tokenId],
+        },
+    ]);
+    return res[0].balance;
+};
+
+module.exports.createClient = () => {
+    return createOmniRpcClient();
+};
+
+module.exports.generate = async function(num = 1) {
+    return createOmniRpcClient().generate(num);
+};
+
+module.exports.activateSegwit = async function() {
+    return createOmniRpcClient().generate(432);
+};
+
+module.exports.swaperoo = async function(aliceDetails, bobDetails, tokenId, omni_value, btc_value) {
+    if (!tokenId) {
+        throw new Error("tokenId must be provided, got: " + tokenId);
+    }
+    // Caveat: If alice has a lot of BTC on her omni output it will all go to Bob
+
+    // alice_output = prev output for omni = new output for BTC (should only have dust BTC)
+    const { alice_keypair, alice_omni_utxo, alice_final_address } = aliceDetails;
+    // bob_btc_output = prev output for BTC = output for BTC change
+    // bob_omni_output = new output for Omni
+    const { bob_keypair, bob_btc_utxo, bob_btc_output, bob_final_address } = bobDetails;
+
+    const alice_output = bitcoin.address.toOutputScript(alice_final_address, regtest);
+    const bob_omni_output = bitcoin.address.toOutputScript(bob_final_address, regtest);
+
+    if (bob_omni_output === bob_btc_output) {
+        throw new Error("Bob BTC and Omni output MUST be different");
+    }
+
+    const txb = new bitcoin.TransactionBuilder();
+
+    const fee = 2500;
+    const dust = 500;
+    const bob_btc = bob_btc_utxo.value - fee / 2 - btc_value - dust;
+    const alice_btc = alice_omni_utxo.value - fee / 2 + btc_value;
+
+    // Alice Omni input
+    txb.addInput(alice_omni_utxo.txid, alice_omni_utxo.vout, null, alice_output);
+    // Bob BTC input
+    txb.addInput(bob_btc_utxo.txid, bob_btc_utxo.vout, null, bob_btc_output);
+
+    // Add BTC change back to Bob
+    txb.addOutput(bob_btc_output, bob_btc);
+    // Add BTC output to Alice
+    txb.addOutput(alice_output, alice_btc);
+    // Add Omni output to Bob (it's Omni because it's different from the inputs)
+    txb.addOutput(bob_omni_output, dust);
+
+    // Add Omni instructions
+    const payload = await createOmniRpcClient().command([
+        {
+            method: "omni_createpayload_simplesend",
+            parameters: [tokenId, omni_value.toString()],
+        },
+    ]);
+    const embed = payloadToEmbed(payload[0]);
+    txb.addOutput(embed.output, 500);
```

Comment:
❓I thought this value should be `5460` as this is the dust level... or did you change the dust level to 500 somewhere?

Reply:
Yes, I did not remember the dust level but as per my odd comment, the behaviour is unexpected, let me do more research.

Reply:
See https://github.com/comit-network/comit-rs/issues/691#issuecomment-462998340 for details